### PR TITLE
router: refactor TestStickyHTTPRouteWebsocket

### PR DESCRIPTION
**Pull request by benburkert**
Use a table to organize the test steps.

Targets flynn/flynn#781

/cc @bgentry

### CODE REVIEW

File: router/http_test.go [hunk 1]
```diff
@@ -410,61 +409,70 @@ func (s *S) TestStickyHTTPRoute(c *C) {
...
+				req.AddCookie(sessionCookie)
+			}
+			req.Header.Set("Connection", "Upgrade")
+			res, err := httpClient.Do(req)
+			defer res.Body.Close()
 
-	discoverdRegisterHTTP(c, l, srv2.Listener.Addr().String())
-	for i := 0; i < 10; i++ {
-		resCookie := assertGetCookieUpgradeClosed(c, "http://"+l.Addr+"/1", "example.com", "ok 1", cookie)
-		c.Assert(resCookie, IsNil)
```
Comment (bgentry) [inline router/http_test.go:461]: This assertion and the similar one below have been lost in this commit. We're not supposed to send back a session cookie unless we're changing the sticky backend (or setting it from nothing).

Comment (benburkert) [inline router/http_test.go:461]: I've brought back this assertion that the cookie is not set unnecessarily.

Comment (bgentry) [inline router/http_test.go:461]: It's still not quite right, as it needs to also make sure that a cookie is set on the final test case (when the original sticky backend has disappeared). May as well do that same assertion on the first test case too so it's easy to see if the cookie isn't getting set at all.

File: router/http_test.go [hunk 2]
```diff
@@ -410,61 +409,77 @@ func (s *S) TestStickyHTTPRoute(c *C) {
...
+	defer discoverd.UnregisterAll()
 
 	addStickyHTTPRoute(c, l)
 
-	discoverdRegisterHTTP(c, l, srv1.Listener.Addr().String())
-	defer discoverd.UnregisterAll()
+	steps := []struct {
+		do          func()
+		backend     string
+		noSetCookie bool
```
Comment (bgentry) [inline router/http_test.go]: I think inverting this and renaming to `shouldSetCookie` or `setCookie` is less confusing.

**Comment by bgentry** [comment]
Great job reorganizing this. Still a beast of a test but it's much more clear what's going on.

**Pull request closed without merging**